test/fulfillment_order_test.rb

```diff
@@ -57,5 +56,91 @@ def setup
+          :body => ActiveSupport::JSON.encode(body)
+
+        response_fulfillment_orders = fulfillment_order.move(new_location_id: new_location_id)
+
+        assert_equal 'closed', fulfillment_order.status
+
+        assert_equal 3, response_fulfillment_orders.count
+        original_fulfillment_order = response_fulfillment_orders['original_fulfillment_order']
+        refute_nil original_fulfillment_order
+        assert original_fulfillment_order.is_a?(ShopifyAPI::FulfillmentOrder)
+        assert_equal 'closed', original_fulfillment_order.status
+
+        moved_fulfillment_order = response_fulfillment_orders['moved_fulfillment_order']
+        refute_nil moved_fulfillment_order
+        assert moved_fulfillment_order.is_a?(ShopifyAPI::FulfillmentOrder)
+        assert_equal 'open', moved_fulfillment_order.status
+        assert_equal new_location_id, moved_fulfillment_order.assigned_location_id
+
+        remaining_fulfillment_order = response_fulfillment_orders['remaining_fulfillment_order']
+        assert_nil remaining_fulfillment_order
+      end
+    end
+
+    context "#cancel" do
+      should "cancel a fulfillment order" do
+        fulfillment_order = ShopifyAPI::FulfillmentOrder.find(519788021)
+        assert_equal 'open', fulfillment_order.status
+
+        cancelled = ActiveSupport::JSON.decode(load_fixture('fulfillment_order'))
+        cancelled['status'] = 'cancelled'
+        body = {
+          fulfillment_order: cancelled,
+          replacement_fulfillment_order: fulfillment_order,
+        }
+        fake "fulfillment_orders/519788021/cancel", :method => :post, :body => ActiveSupport::JSON.encode(body)
+
+        response_fulfillment_orders = fulfillment_order.cancel
+
+        assert_equal 'cancelled', fulfillment_order.status
+        assert_equal 2, response_fulfillment_orders.count
+        fulfillment_order = response_fulfillment_orders['fulfillment_order']
+        assert_equal 'cancelled', fulfillment_order.status
+        replacement_fulfillment_order = response_fulfillment_orders['replacement_fulfillment_order']
+        assert_equal 'open', replacement_fulfillment_order.status
+      end
+    end
+
+    context "#close" do
+      should "be able to close fulfillment order" do
+        fulfillment_order = ShopifyAPI::FulfillmentOrder.find(519788021)
+
+        closed = ActiveSupport::JSON.decode(load_fixture('fulfillment_order'))
+        closed['status'] = 'closed'
+        request_body = {
+          fulfillment_order: {
+            message: "Test close message."
+          }
+        }
+        fake "fulfillment_orders/519788021/close", :method => :post,
+          :request_body => ActiveSupport::JSON.encode(request_body),
+          :body => ActiveSupport::JSON.encode(closed)
+
+        assert_equal 'open', fulfillment_order.status
+        assert fulfillment_order.close(message: "Test close message.")
+        assert_equal 'closed', fulfillment_order.status
```

Comment:
I understand that the call to Shopify core is a stub. Just want to point out that in case of `close`, `fulfillment_order` status has to be `in_progress`. It becomes `incomplete` after the call.